orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/kato/pipeline/support/SourceResolverSpec.groovy

```diff
@@ -89,6 +90,76 @@ class SourceResolverSpec extends Specification {
     "useSourceCapacity"    | "allDisabled"    || "test-v001"     || "test"          || "us-west-1"
   }
 
+  void "should populate deploy stage 'source' with targeted server group if source is absent and target is explicitly specified"() {
+    given:
+    OortService oort = Mock(OortService)
+    ObjectMapper mapper = new ObjectMapper()
+    SourceResolver resolver = new SourceResolver(
+      oortService: oort,
+      mapper: mapper,
+      resolver: new TargetServerGroupResolver(oortService: oort, mapper: mapper)
+    )
+
+    when:
+    def stage = new PipelineStage(
+      new Pipeline(),
+      "test",
+      [
+        application: "app",
+        credentials: "test",
+        region: "us-west-1",
+        target: "current_asg_dynamic",
+        targetCluster: "app-test"
+      ]
+    )
+    def source = resolver.getSource(stage)
+
+    then:
+    1 * oort.getTargetServerGroup(
```

Comment:
Thanks for pointing that out; I forgot it existed. I'll switch over to that later on as it would be a more intrusive change.